modules/ROOT/pages/dataweave-type-selector.adoc

```diff
@@ -0,0 +1,209 @@
+= Type Selector
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+The type selector allows the use of the `.` selector over xref:dataweave-type-system.adoc#composite-types[types] which enables you to easily declare types from existing ones. This feature can also be combined with the new xref:dataweave-selectors-reusing-types.adoc[module loader feature] to load files and translate declarations inside your file into DataWeave type directives which can be accessed in the same way as types from any other module:
+
+Example: DataWeave script
+[source,dataweave,linenums]
+----
+%dw 2.0
+---
+type User = {name: String}
+type NameType = User.name
+var name: NameType = "Seba"
+---
+typeOf(name)
+----
+
+Output 
+----
+"String"
+----
+
+== Examples on How to Use Type Selector
+
+The following example describes how type selection is useful for declaring complex types from other types already declared:
+
+Example: Composition
+
+DataWeave script
```

Comment:
```suggestion
.DataWeave script:

```
That will turn this into a a heading of sorts to distinguish it from other text, such as you see here: https://docs.mulesoft.com/dataweave/2.2/dataweave-cookbook-extract-data#single_value

Or alternatively, you could turn them into H3 headings, such as 
`=== DataWeave Script` under each example or whatever. 

let's discuss formatting options